contrib/gofiber/fiber.v2/fiber_test.go

```diff
@@ -0,0 +1,242 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package fiber
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestChildSpan(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	router := fiber.New()
+	router.Use(Middleware(WithServiceName("foobar")))
+	router.Get("/user/:id", func(c *fiber.Ctx) error {
+		id := c.Params("id")
+		_, ok := tracer.SpanFromContext(c.Context())
+		assert.True(ok)
+
+		return c.SendString(id)
+	})
+
+	r := httptest.NewRequest("GET", "/user/123", nil)
+	resp, err := router.Test(r, 100)
+
+	assert.Equal(nil, err)
+	assert.Equal(resp.StatusCode, 200)
+}
+
+func TestTrace200(t *testing.T) {
+	assertDoRequest := func(assert *assert.Assertions, mt mocktracer.Tracer, router *fiber.App) {
+		r := httptest.NewRequest("GET", "/user/123", nil)
+
+		// do and verify the request
+		resp, err := router.Test(r, 100)
+		assert.Equal(nil, err)
+		assert.Equal(resp.StatusCode, 200)
+
+		// verify traces look good
+		spans := mt.FinishedSpans()
+		assert.Len(spans, 1)
+		if len(spans) < 1 {
+			t.Fatalf("no spans")
+		}
+		span := spans[0]
+		assert.Equal("http.request", span.OperationName())
+		assert.Equal(ext.SpanTypeWeb, span.Tag(ext.SpanType))
+		assert.Equal("foobar", span.Tag(ext.ServiceName))
+		assert.Equal("GET /user/:id", span.Tag(ext.ResourceName))
+		assert.Equal("200", span.Tag(ext.HTTPCode))
+		assert.Equal("GET", span.Tag(ext.HTTPMethod))
+		assert.Equal("/user/123", span.Tag(ext.HTTPURL))
+	}
+
+	t.Run("response written", func(t *testing.T) {
+		assert := assert.New(t)
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		router := fiber.New()
+		router.Use(Middleware(WithServiceName("foobar")))
+		router.Get("/user/:id", func(c *fiber.Ctx) error {
+			span, ok := tracer.SpanFromContext(c.Context())
+			assert.True(ok)
+			assert.Equal(span.(mocktracer.Span).Tag(ext.ServiceName), "foobar")
+			id := c.Params("id")
+
+			return c.SendString(id)
+		})
+		assertDoRequest(assert, mt, router)
+	})
+
+	t.Run("no response written", func(t *testing.T) {
```

Comment:
1. Always use no more than one word in the test name (just like all other tests in the package). I realise other languages have the `it ...` syntax but Go is different. The test name is simply an identifier and is not meant to be a description of the test.
2. Most of the body of this test seems to be shared, can we factor it out into the main test and reuse it?

There are also a couple of weird things here:
* Don't cast `span` to `mocktracer.Span`. Instead use the `mocktracer` to pull the spans you need. That's why you likely get the test failure.
* I don't see any difference between the two tests, but maybe that's just me.

Reply:
@gbbr thanks for the pointers. I'm now seeing an error regarding the current version tag. I'm not sure I'm supposed to update it myself?

Reply:
@andrei-dascalu The version tag error happened because v1.28.0 was just released. If you upmerge to the `v1` branch, that test should pass again.

For test naming we generally prefer one word names, but two words separated with a dash (`-`) are also acceptable. We've recently added a [style guide](https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#tests) to help with this and other style-related things.